docs-src/docs.md

```diff
@@ -0,0 +1,422 @@
+
+Please follow [usage](https://github.com/flogei/ci-demo-walkthrough/blob/master/docs/usage.md) to see Jenkins in action.
+
+## CI/CD Demo Overview
+
+This document assumes you have followed the [setup](https://github.com/flogei/ci-demo-walkthrough/blob/master/docs/setup.md) procedures and have a running Jenkins instance using Docker.
+
+We are going to show a sample process of how CI/CD could look like.
+
+1. Create a new branch for our new feature
+2. A developer builds and tests locally
+3. A developer commits changes to the feature branch
+4. Jenkins builds the branch and pushes it to the respective Apigee Environment
+5. We promote the feature branch to master
+6. Jenkins pushes the branch to Apigee production environment
+
+## Setup for our Feature development 
+
+Right now we have 2 branches: master and prod
+
+![image alt text](images/image_9.png)
+
+You can also check on the command line with
+
+```
+Git branch
+* master
+  prod
+```
+
+As a developer we now want to work on a new feature for our API proxy and therefore we create a new feature branch called *feature/1* from master branch
+
+```
+git checkout -b feature/1
+```
+
+## Local Testing
+
+Let's simulate the steps a proxy developer would take to test their changes before they want a merge request or code review. Ideally the proxy developer can deploy multiple times until all tests pass and they are ready for merge request or code review. For this reason, we don't commit directly to the feature branch-- we don't want to clutter the Git log history, and we also don't want to rely on Jenkins just yet. Deploy and test your changes below, just as a proxy developer would from their local machine:
+
+```
+cd currency-v1
+mvn install -Ptest -Denv.APIGEE_ORG={ORG} -Denv.APIGEE_USERNAME={username} -Denv.APIGEE_PASSWORD={password} -Denv.API_DOMAIN_TEST={apigee_proxy_domain}
+```
+
+Note that -Denv.API_DOMAIN_TEST is the same API_DOMAIN_TEST you've defined in your .run-image.sh file. If you don't want to pass that many arguments to maven, you can add those properties to your settings file *~/.m2/settings.xml*:
+
+```
+<profile>
+    <id>test</id>
+    <properties>
+        <!-- these settings are for maven-jenkins-ci-demo -->
+        <env.APIGEE_ORG>...</env.APIGEE_ORG>
+        <env.APIGEE_USERNAME>...</env.APIGEE_USERNAME>
+        <env.APIGEE_PASSWORD>...</env.APIGEE_PASSWORD>
+        <env.API_DOMAIN_TEST>...</env.API_DOMAIN_TEST>
+    </properties>
+</profile>
+```
+
+If you decide to go with the settings file approach, you can just execute this command instead of the above long maven command:
+
+```
+mvn install -Ptest
+```
+
+Positive
+: Note: If you have a Build failure with a NullPointException you have most likely missed some variables
+
+This command outputs the maven build process and is a good point to look through the different steps taken here:
+
```

Comment:
also linting and unit testing before this starts

you will see `jshint` in the logs near the start and then unit testing is with `mocha`. 

We haven't looked at this alot, but it is mainly used when you have complex javascript for things like logging or crypto operations.

here is the code: https://github.com/apigee/maven-jenkins-ci-demo/blob/master/currency-v1/test/unit/loggly.js

and here is the related article: https://community.apigee.com/articles/3964/unit-testing-javascript-code-with-mocha-sinon-and.html

Reply:
added the Linting and Unit tests part - not sure where to put the other comments in though